src/v/storage/disk_log_impl.cc

```diff
@@ -461,29 +461,37 @@ disk_log_impl::monitor_eviction(ss::abort_source& as) {
 
 model::offset
 disk_log_impl::request_eviction_until_offset(model::offset max_offset) {
-    vlog(
-      gclog.debug,
-      "[{}] requested eviction of segments up to {} offset",
-      config().ntp(),
-      max_offset);
-    // we only notify eviction monitor if there are segments to evict
-    auto have_segments_to_evict
-      = (_segs.size() > 1)
-        && _segs.front()->offsets().get_committed_offset() <= max_offset;
+    const auto min_offset = [this]() -> std::optional<model::offset> {
+        if (_segs.size() > 1) {
+            return _segs.front()->offsets().get_committed_offset();
+        }
+        return std::nullopt;
+    }();
+
+    const auto have_segments_to_evict = min_offset.has_value()
+                                        && min_offset.value() <= max_offset;
 
     if (_eviction_monitor && have_segments_to_evict) {
         _eviction_monitor->promise.set_value(max_offset);
         _eviction_monitor.reset();
 
-        return model::next_offset(max_offset);
-    } else {
         vlog(
-          gclog.debug,
-          "[{}] no segments to evict up to {} offset; skipping eviction",
+          gclog.info,
```

Comment:
Is there a reason for this upgrade to `info` other than this is top of mind?

Reply:
no, it's purely because we see systems out there where we have a mystery and one explanation would be explained by the eviction request not happening.

the caller of this method already logs at info level. lemme see if i can just merge all these together so there is no net-new log lines.

Reply:
nah you're right. these messages can both be at debug level. in all but some extreme cases they are retriggered frequently so we don't need to worry about catching all instances of these logs.